src/main/java/br/unb/cic/analysis/df/OverridingAssignmentAnalysis.java

```diff
@@ -197,4 +238,115 @@ private String getStaticFieldRefName(StaticFieldRef staticFieldRef) {
         return staticFieldRef.getField().getName();
     }
 
+    /*
+     * Returns the name of the JArrayRef
+     */
+    private String getJArrayChain(ValueBox valueBox){
+        String nextKey = ((JArrayRef) valueBox.getValue()).getBase().toString();
+        for (HashMap<String, StaticFieldRef> auxMap : getHashMapStatic()) {
+            for (String mapKey : auxMap.keySet()) {
+                if (mapKey.equals(nextKey)) {
+                    return auxMap.get(mapKey).toString();
+                }
+            }
+        }
+        return "";
+    }
+
+    /*
+     * Returns the generated hashmap chain from a key
+     */
+    private String getJInstanceFieldChain(String nextKey){
```

Comment:
@galilasmb this method is too long and somewhat complex. can it be broken in smaller methods? by extracting them?>

Reply:
explain what are keys, and all the other concepts that you use below

Reply:
Done.